### PR TITLE
Add testing documentation for appointments

### PR DESCRIPTION
**Pull request by sandydays**
As per title

From #64

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -479,3 +479,81 @@ Note: If the format of `visit-now-update` arguments are wrong, the error message
...
+.. Other incorrect appointment delete commands to try: `appt-add`, `appt-delete p/x` (where x is larger than the patient list size) +
+   Expected: Similar to previous.
+
+=== Editing an appointment
+
+. Editing an appointment while all appointments are listed
+
+.. Prerequisites:
+... List all appointments using the `appt-list` command. Multiple appointments in the list.
+... Ensure all the suggested date times below don't already exist in the appointment book. If they do, try with start and end date times that don't exist in the appointment list to get the expected results.
```
Comment (francislow) [inline docs/DeveloperGuide.adoc]: index of appointment must be present is a preq too

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -479,3 +479,81 @@ Note: If the format of `visit-now-update` arguments are wrong, the error message
...
+.. Other incorrect appointment delete commands to try: `appt-edit`, `appt-edit x` (where x is larger than the appointment list size) +
+   Expected: Similar to previous.
+
+=== Deleting a non-recurring appointment
+
+. Deleting a non-recurring appointment while all appointments are listed
+
+.. Prerequisites:
+... List all appointments using the `appt-list` command. Multiple appointments in the list.
+... xth appointment in the appointment list should be a non-recurring appointment, where x is a positive integer.
```
Comment (francislow) [inline docs/DeveloperGuide.adoc]: same here for the index prereq

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -479,3 +479,81 @@ Note: If the format of `visit-now-update` arguments are wrong, the error message
...
+
+.. Prerequisites:
+... List all appointments using the `appt-list` command. Multiple appointments in the list.
+... Ensure all the suggested date times below don't already exist in the appointment book. If they do, try with start and end date times that don't exist in the appointment list to get the expected results.
+.. Test case: `appt-edit 1 p/5 sdt/01-01-2020 1000 edt/01-01-2020 1200` +
+   Expected: First appointment is edited in the list to fifth patient's details from the list of patients and the input start and end date times as mentioned. Details of the edited appointment shown in the status message. Timestamp in the status bar is updated.
+.. Test case: `appt-edit 1 rmon/3` +
+   Expected: First appointment is edited in the list to recur by 3 months. If the appointment was previously not recurring, it will be tagged as `recurring`. Details of the edited appointment shown in the status message. Timestamp in the status bar is updated.
+.. Test case (two commands): `appt-edit 3 p/0` +
+   Expected: No appointment is added. Error details shown in the status message. Status bar remains the same.
```
Comment (francislow) [inline docs/DeveloperGuide.adoc]: I think u forgot this edit this part  "no appointment is added"

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -479,3 +479,81 @@ Note: If the format of `visit-now-update` arguments are wrong, the error message
...
+   Expected: No appointment is deleted. Error details shown in the status message. Status bar remains the same.
+.. Other incorrect appointment delete commands to try: `appt-delete`, `appt-delete x` (where x is larger than the appointment list size) +
+   Expected: Similar to previous.
+
+=== Deleting a recurring appointment
+
+. Deleting a recurring appointment while all appointments are listed
+
+.. Prerequisites:
+... List all appointments using the `appt-list` command. Multiple appointments in the list.
```
Comment (francislow) [inline docs/DeveloperGuide.adoc]: same here

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (sandydays) [reply docs/DeveloperGuide.adoc]: I think by prerequisites they mean the state application, and in this case that of the appointment list, before the command is entered. 

Index of appointment not being after the command word is a test case that throws an error (have mentioned further down).

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (sandydays) [reply docs/DeveloperGuide.adoc]: Thanks for pointing that out!

**Comment by francislow** [comment]
LGTM

**Pull request merged**